### PR TITLE
Update NTVS IoT Extension for next release

### PR DESCRIPTION
**Pull request by munyirik**
This PR includes fixes for:
1. https://github.com/ms-iot/ntvsiot/issues/54
2. https://github.com/ms-iot/ntvsiot/issues/52
3. https://github.com/ms-iot/ntvsiot/issues/49
4. https://github.com/ms-iot/ntvsiot/issues/26

Other updates and fixes:
1. Added msvsmon bootstraping to project system
2. Fixes to CopyNodeChakra PS script to copy Node.exe to device.
3. Updated instructions in project template source code.

### CODE REVIEW

**Comment by munyirik** [comment]
@alanch-ms @bfjelds could you please CR? I need 2 sign-offs.

File: Product/ProjectFlavor/NodejsUwpProjectFlavorCfg.cs [hunk 1]
```diff
@@ -522,127 +504,191 @@ int IVsDeployableProjectCfg.Rollback(uint dwReserved)
...
+        }
+
+        private string GetProjectUniqueName()
+        {
+            EnvDTE.DTE dte = (EnvDTE.DTE)Package.GetGlobalService(typeof(EnvDTE.DTE));
+            EnvDTE.SolutionBuild sb = dte.Solution.SolutionBuild;
+            string projectUniqueName = string.Empty;
+            foreach (String s in (Array)sb.StartupProjects)
+            {
+                // There should only be one startup project so get the first one we find.
```
Comment (alanch-ms) [inline Product/ProjectFlavor/NodejsUwpProjectFlavorCfg.cs]: According to the PTVS folks, this is not true.  A solution can have multiple startup projects. Right click on solution node->Set startup project.

File: Product/ProjectTemplates/CylonUwp/app.js [hunk 2]
```diff
@@ -1,6 +1,10 @@
 // !!BEFORE YOU RUN THIS CODE!!
 
-// 1. Follow the steps in the "Build Serialport" section in the link below:
+// 1. Connect to your device using PowerShell or SSH (see http://windowsondevices.com for a guide to do that) and run:
```
Comment (alanch-ms) [inline Product/ProjectTemplates/CylonUwp/app.js]: This doesn't seem like a very good user experiences.  Imagine you need to deploy a node app to many devices.

Comment (munyirik) [inline Product/ProjectTemplates/CylonUwp/app.js]: Agreed, it's not ideal. I've started another discussion about to explore more options.

File: Product/ProjectFlavor/NodejsUwpProjectFlavorCfg.cs (hunk 1 above)
Comment (munyirik) [inline Product/ProjectFlavor/NodejsUwpProjectFlavorCfg.cs]: @alanch-ms does that apply when the target is a remote machine as well? how does PTVS handle this?

Comment (alanch-ms) [inline Product/ProjectFlavor/NodejsUwpProjectFlavorCfg.cs]: PTVS had a different way to get the unique name.  See the pull request https://github.com/Microsoft/PTVS/pull/954.

Comment (munyirik) [inline Product/ProjectFlavor/NodejsUwpProjectFlavorCfg.cs]: I thought that's the implementation I had before which had a bug. I'll use the PTVS implementation and test. The bug I had before was a problem when there was another project in the solution that was above the startup project.

**Comment by munyirik** [comment]
I think this is ready to be merged. @alanch-ms I've removed the extra step that required running reg.exe to allow xcopy. I'll update PR https://github.com/ms-iot/node-uwp-wrapper/pull/22 shortly.

**Comment by alanch-ms** [comment]
:thumbsup:

**Pull request merged**